libs/ddhub-client-gateway-frontend/ui/core/src/components/Table/Table.effects.ts

```diff
@@ -163,6 +166,32 @@ export function useTableEffects<T>({
     return `Showing ${props.from} to ${(rows.length < props.to ? rows.length : props.to)} of ${rows.length}`
   };
 
+  const handleCheckboxClick = (event: React.MouseEvent<unknown>, name: string) => {
+    const selectedIndex = selected.indexOf(name);
+    let newSelected: string[] = [];
+
+    if (selectedIndex === -1) {
```

Comment:
I suppose here it should also work this:
```
const index = selected.indexOf(name);
if (index > -1) {
  newSelected = [...selected].splice(index, 1)] // we need new array, otherwise we would mutate selected array.
} else {
  newSelected = [...selected, name];
}
```
less cases and easier to read.
or just move it to a function and return new array and later do something like:
```
 const newSelected = createNewList(selected, name) // selected - old list, name it's checkbox name.
```

What do you think?

Reply:
Agreed that this solution is more elegant. Tried it already but it somehow did not work even though the code is correct. Will debug if have time.

Reply:
i made a mistake with code example. This should work:
```
const newArray = (array, value) => {
const index = array.indexOf(name);
  if (index > -1) {
    const copy = [...array];
    copy.splice(index, 1);
    return copy;
  } else {
    return [...array, value]
  }
}
```